sdk/core/System.ClientModel/src/Convenience/ClientCache.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A cache for storing client instances with a Least Recently Used (LRU) eviction policy.
+/// </summary>
+public class ClientCache
```

Comment:
Was `public` intentional here?

Reply:
Yes - https://github.com/Azure/azure-sdk-for-net/pull/48245#discussion_r1953011681